### ISSUE TITLE
Parser having problem with building trees

### ISSUE THREAD

**Issue by NeverInAsh** [coursera-dl/coursera-dl#485]
🚨Please review the [Troubleshooting](../#troubleshooting) section
before reporting any issue. Don't forget to check also the current issues to
avoid duplicates.

### Subject of the issue
Parser having problem with building tree

### Your environment
* Operating System (name/version):Ubuntu 15.10
* Python version:2.7
* coursera-dl version: Github(latest pull as on 18th may )

### Steps to reproduce
Tell us how to reproduce this issue, please don't forget to include
enough information so that you can help us help you:
The same setting works absolutely fine with other courses.
Should lxml be used instead of html5lib (tree parsing problem)

* Is the problem happening with the latest version of the script?
* Do you have all the recommended versions of the modules? See them in the
  file `requirements.txt`.
Yes
* What is the course that you are trying to access?
intro-to-big-data
* What is the precise command line that you are using (don't forget to obfuscate
  your username and password, but leave all other information untouched).
~coursera-dl ./coursera-dl -u username -p password course-name
* What are the precise messages that you get? Please, use the `--debug`
  option before posting the messages as a bug report. Please, copy and paste
  them.  Don't reword/paraphrase the messages.

### Expected behaviour
Tell us what should happen.

### Actual behaviour
Tell us what happens instead. If the script fails, please copy the *entire*
output of the command or the stacktrace (don't forget to obfuscate your
username and password). If you cannot copy the exception, attach a screenshot.

Downloading class: intro-to-big-data
Starting new HTTPS connection (1): class.coursera.org
404 Client Error: Not Found for url: https://class.coursera.org/intro-to-big-data
Starting new HTTPS connection (1): www.coursera.org
Logged in on coursera.org.
Downloaded https://www.coursera.org/api/opencourse.v1/course/intro-to-big-data (38527 bytes)
Parsing syllabus of on-demand course. This may take some time, be patient ...
Gathering video URLs for video_id <wGlXMdGrEeWHOA7t0uQ0Ww>.
Proceeding with download of resolution 540p of <wGlXMdGrEeWHOA7t0uQ0Ww>.
Gathering subtitle URLs for video_id <wGlXMdGrEeWHOA7t0uQ0Ww>.
Gathering transcript URLs for video_id <wGlXMdGrEeWHOA7t0uQ0Ww>.
Gathering supplement URLs for element_id <JaOFB>.
Traceback (most recent call last):
  File "./coursera-dl", line 6, in <module>
    coursera_dl.main()
  File "/home/darkrider19/coursera-dl/coursera/coursera_dl.py", line 1056, in main
    if download_class(args, class_name):
  File "/home/darkrider19/coursera-dl/coursera/coursera_dl.py", line 1034, in download_class
    return download_on_demand_class(args, class_name)
  File "/home/darkrider19/coursera-dl/coursera/coursera_dl.py", line 987, in download_on_demand_class
    args.video_resolution)
  File "/home/darkrider19/coursera-dl/coursera/coursera_dl.py", line 338, in parse_on_demand_syllabus
    lecture['id'])
  File "/home/darkrider19/coursera-dl/coursera/api.py", line 338, in extract_links_from_supplement
    supplement_content, self._extract_links_from_text(value))
  File "/home/darkrider19/coursera-dl/coursera/api.py", line 433, in _extract_links_from_text
    supplement_links = self._extract_links_from_a_tags_in_text(text)
  File "/home/darkrider19/coursera-dl/coursera/api.py", line 493, in _extract_links_from_a_tags_in_text
    soup = BeautifulSoup(text)
  File "/home/darkrider19/coursera-dl/coursera/utils.py", line 18, in <lambda>
    BeautifulSoup = lambda page: BeautifulSoup_(page, 'html5lib')
  File "/home/darkrider19/anaconda2/lib/python2.7/site-packages/bs4/__init__.py", line 156, in __init__
    % ",".join(features))
bs4.FeatureNotFound: Couldn't find a tree builder with the features you requested: html5lib. Do you need to install a parser library?

**Comment by balta2ar** [comment]
Could you try hacking the code and replacing 'html5lib' with 'lxml' and see whether it works?

This line:
File "/home/darkrider19/coursera-dl/coursera/utils.py", line 18, in
BeautifulSoup = lambda page: BeautifulSoup(page, 'html5lib')

**Comment by balta2ar** [comment]
@NeverInAsh Are you sure you have installed everything from requirements.txt?

**Comment by yielder** [comment]
I have the similar issue with building trees. Switch to 'lxml' not work as well.
Operating System (name/version):Ubuntu 14.04
Python version:3.4

here's the output after switch to lxml
```
Downloaded https://www.coursera.org/api/opencourse.v1/course/progfun2 (16728 bytes)
Parsing syllabus of on-demand course. This may take some time, please be patient ...
Processing module  for-expressions-and-monads
Processing section     recap-getting-started-with-tools
Processing lecture         tools-setup-for-linux
Processing lecture         tools-setup-for-mac-os-x
Processing lecture         tools-for-setup-for-windows
Processing lecture         tutorial-working-on-the-programming-assignments
Processing lecture         troubleshooting
Traceback (most recent call last):
  File "./coursera-dl", line 6, in <module>
    coursera_dl.main()
  File "/mnt/data/coursera/coursera-dl/coursera/coursera_dl.py", line 217, in main
    if download_class(args, class_name):
  File "/mnt/data/coursera/coursera-dl/coursera/coursera_dl.py", line 194, in download_class
    return download_on_demand_class(args, class_name)
  File "/mnt/data/coursera/coursera-dl/coursera/coursera_dl.py", line 130, in download_on_demand_class
    args.video_resolution)
  File "/mnt/data/coursera/coursera-dl/coursera/extractors.py", line 52, in get_modules
    subtitle_language, video_resolution)
  File "/mnt/data/coursera/coursera-dl/coursera/extractors.py", line 130, in _parse_on_demand_syllabus
    lecture[id])
  File "/mnt/data/coursera/coursera-dl/coursera/api.py", line 516, in extract_links_from_supplement
    instructions = (IN_MEMORY_MARKER + self._prettify_instructions(value),
  File "/mnt/data/coursera/coursera-dl/coursera/api.py", line 183, in _prettify_instructions
    self._convert_instructions_basic(soup)
  File "/mnt/data/coursera/coursera-dl/coursera/api.py", line 197, in _convert_instructions_basic
    soup.head.append(css_soup)
AttributeError: NoneType object has no attribute append
```

**Comment by balta2ar** [comment]
This should be fixed in 0.7.0. Closing.
Feel free to comment if it's still actual.
